packages/edit-site/src/components/global-styles/style-variations-container.js

```diff
@@ -29,28 +41,84 @@ export default function StyleVariationsContainer( { gap = 2 } ) {
 		);
 	} );
 
-	const withEmptyVariation = useMemo( () => {
-		return [
+	const themeVariations = useMemo( () => {
+		const withEmptyVariation = [
 			{
 				title: __( 'Default' ),
 				settings: {},
 				styles: {},
 			},
-			...( multiplePropertyVariations ?? [] ).map( ( variation ) => ( {
-				...variation,
-				settings: variation.settings ?? {},
-				styles: variation.styles ?? {},
-			} ) ),
+			...( multiplePropertyVariations ?? [] ),
+		];
+		return [
+			...withEmptyVariation.map( ( variation ) => {
+				const blockStyles = { ...variation?.styles?.blocks } || {};
+				// We need to copy any user custom CSS to the variation to prevent it being lost
+				// when switching variations.
+				if ( userStyles?.blocks ) {
+					Object.keys( userStyles.blocks ).forEach( ( blockName ) => {
+						// First get any block specific custom CSS from the current user styles and merge with any custom CSS for
+						// that block in the variation.
+						if ( userStyles.blocks[ blockName ].css ) {
+							const variationBlockStyles =
+								blockStyles[ blockName ] || {};
+							const customCSS =
+								blockStyles[ blockName ]?.css ||
+								userStyles.blocks[ blockName ].css
+									? {
+											css: `${
+												blockStyles[ blockName ]?.css ||
+												''
+											} ${
+												userStyles.blocks[ blockName ]
+													.css || ''
+											}`,
+									  }
+									: {};
+
+							blockStyles[ blockName ] = {
+								...variationBlockStyles,
+								...customCSS,
+							};
+						}
+					} );
+				}
+				// Now merge any global custom CSS from current user styles with global custom CSS in the variation.
+				const css =
+					userStyles?.css || variation.styles?.css
+						? {
+								css: `${ variation.styles?.css || '' } ${
+									userStyles?.css || ''
+								}`,
+						  }
+						: {};
+
+				const blocks =
+					Object.keys( blockStyles ).length > 0
+						? { blocks: blockStyles }
+						: {};
+
+				const styles = {
+					...variation.styles,
+					...css,
+					...blocks,
+				};
+				return {
+					...variation,
+					settings: variation.settings ?? {},
+					styles: styles ?? {},
```

Comment:
I'm not sure `styles` will ever be falsy as it's destructuring `css` above, and `css` will always be an empty object I think.

Reply:
True, thanks, fixed